[Bug]: Issue Running Whisper with vllm/vllm-openai:v0.10.2 (Works Fine on v0.10.1)

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
/usr/local/lib/python3.12/dist-packages/torch/cuda/__init__.py:63: FutureWarning: The pynvml package is deprecated. Please install nvidia-ml-py instead. If you did not install pynvml directly, please report this to the maintainers of the package that installed pynvml for you.
  import pynvml  # type: ignore[import]
Collecting environment information...
==============================
        System Info
==============================
OS                           : Ubuntu 22.04.5 LTS (x86_64)
GCC version                  : (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version                : Could not collect
CMake version                : version 4.1.0
Libc version                 : glibc-2.35

==============================
       PyTorch Info
==============================
PyTorch version              : 2.8.0+cu128
Is debug build               : False
CUDA used to build PyTorch   : 12.8
ROCM used to build PyTorch   : N/A

==============================
      Python Environment
==============================
Python version               : 3.12.11 (main, Jun  4 2025, 08:56:18) [GCC 11.4.0] (64-bit runtime)
Python platform              : Linux-5.15.0-141-generic-x86_64-with-glibc2.35

==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : 12.8.93
CUDA_MODULE_LOADING set to   : LAZY
GPU models and configuration : GPU 0: NVIDIA A100-SXM4-80GB
Nvidia driver version        : 560.35.03
cuDNN version                : Could not collect
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True

==============================
          CPU Info
==============================
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        45 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               32
On-line CPU(s) list:                  0-31
Vendor ID:                            AuthenticAMD
Model name:                           AMD EPYC 7763 64-Core Processor
CPU family:                           25
Model:                                1
Thread(s) per core:                   1
Core(s) per socket:                   1
Socket(s):                            32
Stepping:                             1
BogoMIPS:                             4899.99
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl tsc_reliable nonstop_tsc cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext invpcid_single ibpb vmmcall fsgsbase bmi1 avx2 smep bmi2 invpcid rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves clzero wbnoinvd arat umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor
Hypervisor vendor:                    VMware
Virtualization type:                  full
L1d cache:                            1 MiB (32 instances)
L1i cache:                            1 MiB (32 instances)
L2 cache:                             16 MiB (32 instances)
L3 cache:                             1 GiB (32 instances)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-31
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Mitigation; safe RET
Vulnerability Spec store bypass:      Vulnerable
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Retpolines; IBPB conditional; STIBP disabled; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

==============================
Versions of relevant libraries
==============================
[pip3] flashinfer-python==0.3.0
[pip3] numpy==2.2.6
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cudnn-frontend==1.14.1
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-cufile-cu12==1.13.1.3
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-ml-py==13.580.82
[pip3] nvidia-nccl-cu12==2.27.3
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] pynvml==13.0.1
[pip3] pyzmq==27.1.0
[pip3] torch==2.8.0+cu128
[pip3] torchaudio==2.8.0+cu128
[pip3] torchvision==0.23.0+cu128
[pip3] transformers==4.56.1
[pip3] triton==3.4.0
[conda] Could not collect

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
vLLM Version                 : 0.10.2
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled
GPU Topology:
  	GPU0	CPU Affinity	NUMA Affinity	GPU NUMA ID
GPU0	 X 	0-31	0		N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

==============================
     Environment Variables
==============================
NVIDIA_VISIBLE_DEVICES=all
NVIDIA_REQUIRE_CUDA=cuda>=12.8 brand=unknown,driver>=470,driver<471 brand=grid,driver>=470,driver<471 brand=tesla,driver>=470,driver<471 brand=nvidia,driver>=470,driver<471 brand=quadro,driver>=470,driver<471 brand=quadrortx,driver>=470,driver<471 brand=nvidiartx,driver>=470,driver<471 brand=vapps,driver>=470,driver<471 brand=vpc,driver>=470,driver<471 brand=vcs,driver>=470,driver<471 brand=vws,driver>=470,driver<471 brand=cloudgaming,driver>=470,driver<471 brand=unknown,driver>=535,driver<536 brand=grid,driver>=535,driver<536 brand=tesla,driver>=535,driver<536 brand=nvidia,driver>=535,driver<536 brand=quadro,driver>=535,driver<536 brand=quadrortx,driver>=535,driver<536 brand=nvidiartx,driver>=535,driver<536 brand=vapps,driver>=535,driver<536 brand=vpc,driver>=535,driver<536 brand=vcs,driver>=535,driver<536 brand=vws,driver>=535,driver<536 brand=cloudgaming,driver>=535,driver<536 brand=unknown,driver>=550,driver<551 brand=grid,driver>=550,driver<551 brand=tesla,driver>=550,driver<551 brand=nvidia,driver>=550,driver<551 brand=quadro,driver>=550,driver<551 brand=quadrortx,driver>=550,driver<551 brand=nvidiartx,driver>=550,driver<551 brand=vapps,driver>=550,driver<551 brand=vpc,driver>=550,driver<551 brand=vcs,driver>=550,driver<551 brand=vws,driver>=550,driver<551 brand=cloudgaming,driver>=550,driver<551 brand=unknown,driver>=560,driver<561 brand=grid,driver>=560,driver<561 brand=tesla,driver>=560,driver<561 brand=nvidia,driver>=560,driver<561 brand=quadro,driver>=560,driver<561 brand=quadrortx,driver>=560,driver<561 brand=nvidiartx,driver>=560,driver<561 brand=vapps,driver>=560,driver<561 brand=vpc,driver>=560,driver<561 brand=vcs,driver>=560,driver<561 brand=vws,driver>=560,driver<561 brand=cloudgaming,driver>=560,driver<561 brand=unknown,driver>=565,driver<566 brand=grid,driver>=565,driver<566 brand=tesla,driver>=565,driver<566 brand=nvidia,driver>=565,driver<566 brand=quadro,driver>=565,driver<566 brand=quadrortx,driver>=565,driver<566 brand=nvidiartx,driver>=565,driver<566 brand=vapps,driver>=565,driver<566 brand=vpc,driver>=565,driver<566 brand=vcs,driver>=565,driver<566 brand=vws,driver>=565,driver<566 brand=cloudgaming,driver>=565,driver<566
NCCL_VERSION=2.25.1-1
NVIDIA_DRIVER_CAPABILITIES=compute,utility
NVIDIA_PRODUCT_NAME=CUDA
VLLM_USAGE_SOURCE=production-docker-image
CUDA_VERSION=12.8.1
LD_LIBRARY_PATH=/usr/local/cuda/lib64
CUDA_HOME=/usr/local/cuda
CUDA_HOME=/usr/local/cuda
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1
CUDA_MODULE_LOADING=LAZY
```

</details>


### 🐛 Describe the bug

I tried serving the Whisper model, but whenever I send the following request, I encounter an error. Moreover, the request seems to bring the whole model down.
```
audio_file_path  = file_path + "/../data/test_files/d6_10_20_17-625ddc96edfd375a7cbfa57189dfb064-2905E5.wav"

openai_api_key = "EMPTY"
openai_api_base = "http://0.0.0.0:8000/v1"
client = OpenAI(api_key=openai_api_key, base_url=openai_api_base)
sync_openai(audio_file_path, client)

def sync_openai(audio_file_path: str, client: OpenAI):
    t1 = time.time()
    with open(audio_file_path, "rb") as f:
        transcription = client.audio.transcriptions.create(
            file=f,
            model="whisper",
            language="fa",
            response_format="json",
            temperature=0.0,
            # Additional sampling params not provided by OpenAI API.
            extra_body=dict(
                seed=4419,
                # repetition_penalty=1.3,
            ),
        )
    t2 = time.time()
    print(t2-t1)
    print("transcription result:", transcription.text)
```
In a way, this request acts like a "kill request" for the model.

I’m running Docker with the image vllm/vllm-openai:v0.10.2 and I serve the model using the following command:
```
vllm serve /app/data/asr/whisper \
  --gpu-memory-utilization 0.06 \
  --dtype bfloat16 \
  --block-size 16 \
  --max-num-seqs 32 \
  --served-model-name whisper \
  --middleware src.timing_middleware.log_timing_middleware
```
The client throws the following error:
```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/home/hediehloo/myenvs/globalenv/lib/python3.12/site-packages/openai/_utils/_utils.py", line 287, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/hediehloo/myenvs/globalenv/lib/python3.12/site-packages/openai/resources/audio/transcriptions.py", line 333, in create
    return self._post(  # type: ignore[return-value]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hediehloo/myenvs/globalenv/lib/python3.12/site-packages/openai/_base_client.py", line 1256, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hediehloo/myenvs/globalenv/lib/python3.12/site-packages/openai/_base_client.py", line 1044, in request
    raise self._make_status_error_from_response(err.response) from None
openai.InternalServerError: Error code: 500 - {'error': {'message': 'EngineCore encountered an issue. See stack trace (above) for the root cause.', 'type': 'Internal Server Error', 'param': None, 'code': 500}}
```
Interestingly, when I switch back to vllm/vllm-openai:v0.10.1, everything works perfectly without any issues.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

@DarkLight1337 
```
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720] EngineCore encountered a fatal error.
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720] Traceback (most recent call last):
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 711, in run_engine_core
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]     engine_core.run_busy_loop()
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 738, in run_busy_loop
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]     self._process_engine_step()
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 764, in _process_engine_step
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]     outputs, model_executed = self.step_fn()
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]                               ^^^^^^^^^^^^^^
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 291, in step
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]     scheduler_output = self.scheduler.schedule()
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]                        ^^^^^^^^^^^^^^^^^^^^^^^^^
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/core/sched/scheduler.py", line 464, in schedule
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]     assert ("whisper"
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720]             ^^^^^^^^^
whisper  | (EngineCore_DP0 pid=225) ERROR 09-21 14:21:46 [core.py:720] AssertionError: Whisper is the only supported encoder-decoder model.
whisper  | (APIServer pid=28) ERROR 09-21 14:21:46 [async_llm.py:485] AsyncLLM output_handler failed.
whisper  | (APIServer pid=28) ERROR 09-21 14:21:46 [async_llm.py:485] Traceback (most recent call last):
whisper  | (APIServer pid=28) ERROR 09-21 14:21:46 [async_llm.py:485]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/async_llm.py", line 444, in output_handler
whisper  | (APIServer pid=28) ERROR 09-21 14:21:46 [async_llm.py:485]     outputs = await engine_core.get_output_async()
whisper  | (APIServer pid=28) ERROR 09-21 14:21:46 [async_llm.py:485]               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
whisper  | (APIServer pid=28) ERROR 09-21 14:21:46 [async_llm.py:485]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core_client.py", line 845, in get_output_async
whisper  | (APIServer pid=28) ERROR 09-21 14:21:46 [async_llm.py:485]     raise self._format_exception(outputs) from None
whisper  | (APIServer pid=28) ERROR 09-21 14:21:46 [async_llm.py:485] vllm.v1.engine.exceptions.EngineDeadError: EngineCore encountered an issue. See stack trace (above) for the root cause.
whisper  | (APIServer pid=28) INFO:     172.18.160.73:40040 - "POST /v1/audio/transcriptions HTTP/1.1" 500 Internal Server Error
whisper  | (EngineCore_DP0 pid=225) Process EngineCore_DP0:
whisper  | (EngineCore_DP0 pid=225) Traceback (most recent call last):
whisper  | (EngineCore_DP0 pid=225)   File "/usr/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
whisper  | (EngineCore_DP0 pid=225)     self.run()
whisper  | (EngineCore_DP0 pid=225)   File "/usr/lib/python3.12/multiprocessing/process.py", line 108, in run
whisper  | (EngineCore_DP0 pid=225)     self._target(*self._args, **self._kwargs)
whisper  | (EngineCore_DP0 pid=225)   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 722, in run_engine_core
whisper  | (EngineCore_DP0 pid=225)     raise e
whisper  | (EngineCore_DP0 pid=225)   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 711, in run_engine_core
whisper  | (EngineCore_DP0 pid=225)     engine_core.run_busy_loop()
whisper  | (EngineCore_DP0 pid=225)   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 738, in run_busy_loop
whisper  | (EngineCore_DP0 pid=225)     self._process_engine_step()
whisper  | (EngineCore_DP0 pid=225)   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 764, in _process_engine_step
whisper  | (EngineCore_DP0 pid=225)     outputs, model_executed = self.step_fn()
whisper  | (EngineCore_DP0 pid=225)                               ^^^^^^^^^^^^^^
whisper  | (EngineCore_DP0 pid=225)   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 291, in step
whisper  | (EngineCore_DP0 pid=225)     scheduler_output = self.scheduler.schedule()
whisper  | (EngineCore_DP0 pid=225)                        ^^^^^^^^^^^^^^^^^^^^^^^^^
whisper  | (EngineCore_DP0 pid=225)   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/core/sched/scheduler.py", line 464, in schedule
whisper  | (EngineCore_DP0 pid=225)     assert ("whisper"
whisper  | (EngineCore_DP0 pid=225)             ^^^^^^^^^
whisper  | (EngineCore_DP0 pid=225) AssertionError: Whisper is the only supported encoder-decoder model.
whisper  | [rank0]:[W921 14:21:46.332756937 ProcessGroupNCCL.cpp:1538] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
whisper  | (APIServer pid=28) INFO:     172.18.160.73:40040 - "POST /v1/audio/transcriptions HTTP/1.1" 500 Internal Server Error
whisper  | (APIServer pid=28) INFO:     172.18.160.73:40040 - "POST /v1/audio/transcriptions HTTP/1.1" 500 Internal Server Error
whisper  | (APIServer pid=28) INFO:     Shutting down
whisper  | (APIServer pid=28) INFO:     Waiting for application shutdown.
whisper  | (APIServer pid=28) INFO:     Application shutdown complete.
whisper  | (APIServer pid=28) INFO:     Finished server process [28]
whisper  | (APIServer pid=28) 
```

--- a/vllm/v1/core/sched/scheduler.py
+++ b/vllm/v1/core/sched/scheduler.py
@@ -463,10 +463,6 @@ class Scheduler(SchedulerInterface):
                     # always padded to the maximum length. If we support other
                     # encoder-decoder models, this will need to be updated if we
                     # want to only allocate what is needed.
-                    assert ("whisper"
-                            in self.vllm_config.model_config.model.lower()), (
-                                "Whisper is the only supported "
-                                "encoder-decoder model.")
                     num_encoder_tokens =\
                         self.scheduler_config.max_num_encoder_input_tokens
                 else:

